web/src/narrow_state.js

```diff
@@ -126,6 +126,14 @@ export function stream_name() {
     return undefined;
 }
 
+export function stream_id() {
+    const stream = stream_name();
+    if (stream === undefined) {
+        return undefined;
+    }
+    return stream_data.get_stream_id(stream);
+}
```

Comment:
I would prefer for this to perhaps call `stream_sub` and access `.id` if present -- we're in the process of migrating away from the `stream_name` being a core part of how we reference data in the codebase, and doing so avoids adding another copy of `stream_name => id` lookup logic.